### PR TITLE
Fix default parameters ovverride

### PR DESCRIPTION
**Pull request by Barrett17**
This fixes issue #114.

* Hardcoded parameters have been removed, the train command now defaults to mnist dataset, automatically.
* Parameters are now chosen depending on what the user specify in the command line.

### CODE REVIEW

**Comment by hughperkins** [comment]
@merceyz Thoughts?

File: src/main/train.cpp [hunk 1]
```diff
@@ -495,6 +495,55 @@ void printUsage(char *argv[], Config config) {
     // [[[end]]]
 }
 
+int setDefaults(Config& config, string dataset) {
+    string dataDir;
+    string trainFile;
+    string validateFile;
+    int loadOnDemand = 0;
+
+    if(dataset == "mnist" || dataset.empty()) {
```
Comment (merceyz) [inline src/main/train.cpp]: It shouldn't check dataset.empty() like this.
I see that you did this to set the default values in `Config`, but when dataset isn't used this data isn't needed, thus this doesn't need to be set.

File: src/main/train.cpp [hunk 2]
```diff
@@ -591,38 +640,15 @@ int main(int argc, char *argv[]) {
         }
     }
     string dataset = toLower(config.dataset);
-    if(dataset != "") {
-        if(dataset == "mnist") {
```
Comment (merceyz) [inline src/main/train.cpp]: I'd suggest moving your changes from `setDefaults` to here.

File: src/main/train.cpp [hunk 3]
```diff
@@ -591,38 +640,15 @@ int main(int argc, char *argv[]) {
...
-        cout << "Using dataset " << dataset << ":" << endl;
-        cout << "   datadir: " << config.dataDir << ":" << endl;
-        cout << "   trainfile: " << config.trainFile << ":" << endl;
-        cout << "   validatefile: " << config.validateFile << ":" << endl;
-    }
+
+    if(setDefaults(config, dataset) == -1)
+      return -1;
+
+    cout << "Using dataset " << dataset << ":" << endl;
```
Comment (merceyz) [inline src/main/train.cpp]: You've made it so that it will always print this info, even if a dataset isn't used.

**Comment by merceyz** [comment]
@hughperkins Comments made inline

**Comment by Barrett17** [comment]
Thanks for review, I just tried to mimic the old behavior with my changes. The commit should be OK now.

**Comment by hughperkins** [comment]
Thank you Barrett

**Pull request merged**